docs/autogen.py

```diff
@@ -617,15 +590,34 @@ def render_function(function, method=True):
     subblocks = []
     signature = get_function_signature(function, method=method)
     signature = signature.replace(function.__module__ + '.', '')
-    level = 3
-    subblocks.append('#' * level + ' ' + function.__name__ + '\n')
+    subblocks.append('### ' + function.__name__ + '\n')
     subblocks.append(code_snippet(signature))
     docstring = function.__doc__
     if docstring:
         subblocks.append(process_docstring(docstring))
     return '\n\n'.join(subblocks)
 
 
+def read_page_data(page_data, type):
+    assert type in ['classes', 'functions']
+    data = page_data.get(type, [])
+    for module in page_data.get('all_module_{}'.format(type), []):
+        module_data = []
+        for name in dir(module):
+            if name[0] == '_' or name in EXCLUDE:
+                continue
+            module_member = getattr(module, name)
+            if (inspect.isclass(module_member) and type == 'classes' or
+               inspect.isfunction(module_member) and type == 'functions'):
+                instance = module_member
+                if module.__name__ in instance.__module__:
```

Comment:
I also think this will work just fine. I tested it with the current doc and everything is ok.